client/client_core.go

```diff
@@ -0,0 +1,57 @@
+package client
+
+import (
+	"context"
+	contextUtils "github.com/firebolt-db/firebolt-go-sdk/context"
+
+	"github.com/firebolt-db/firebolt-go-sdk/types"
+
+	"github.com/firebolt-db/firebolt-go-sdk/logging"
+)
+
+type ClientImplCore struct {
+	AccountName string
+	BaseClient
+}
+
+func MakeClientCore(settings *types.FireboltSettings) (*ClientImplCore, error) {
+	client := &ClientImplCore{
+		BaseClient: BaseClient{
+			ApiEndpoint: settings.Url,
+			UserAgent:   ConstructUserAgentString(),
+		},
+		AccountName: settings.AccountName,
+	}
+	client.ParameterGetter = client.GetQueryParams
+	client.AccessTokenGetter = client.getAccessToken
+
+	if err := initialiseCaches(); err != nil {
+		logging.Infolog.Printf("Error during cache initialisation: %v", err)
+	}
+
+	return client, nil
+}
+
+func (c *ClientImplCore) getOutputFormat(ctx context.Context) string {
+	if contextUtils.IsStreaming(ctx) {
+		return jsonLinesOutputFormat
+	}
+	return jsonOutputFormat
+}
+
+func (c *ClientImplCore) GetQueryParams(ctx context.Context, setStatements map[string]string) (map[string]string, error) {
+	params := map[string]string{"output_format": c.getOutputFormat(ctx)}
+	for setKey, setValue := range setStatements {
+		params[setKey] = setValue
+	}
+	return params, nil
+}
```

Comment:
Can these two be moved to the `BaseClient` since they're the same for core/non-core implementations?